NewForce-Cohort-5/tabloidmvc-the-daily-debuggers
# Bg view user profiles

# Description

Allows admins to view a list of user profiles when you click users from the top of the page. Users with only author access will not be able to view the list and will be sent to the home page when trying to view them.  Fixed a merge error in CategoryController.

## Type of change


# Testing Instructions

1. Clone down the repository.
2. `git fetch -a`
3. `git checkout bg-ViewUserProfiles`
4. Navigate to the root directory for this repo (directory-to)/tabloidmvc-the-daily-debuggers
5. Start TabloidMVC.sln
6. Make sure your db is created with seed data. Run [01_TabloidMVC_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-daily-debuggers/blob/main/SQL/01_TabloidMVC_Create_DB.sql) and [02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-daily-debuggers/blob/main/SQL/02_TabloidMVC_Seed_Data.sql) queries in Visual Studio.
7. Start a query on the database called TabloidMVC.
8. Inside this query add the following insert statements:
`INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId) VALUES ('Andy', 'Andy', 'Birt', 'abirt@mail.com', GETDATE(), null, 2)`
`INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId) VALUES ('Nick', 'Nick', 'Perry', 'nperry@mail.com', GETDATE(), null, 2)`
`INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId) VALUES ('Brent', 'Brent', 'Griffith', 'bgriffith@mail.com', GETDATE(), null, 2)`
And run it.
10. Start the server.
11. Log in as admin [admin@example.com](mailto:admin@example.com) (if not already logged in as admin).
12. Click Users at the top of the page
13. You should see a list of all users.
14. Logout and login as one of the accounts you created in step 8 (abirt@mail.com)
15. Click Users at the top of the page
16. You should have been redirected back to the home page as author accounts (such as the ones created in step 8) do not have access to the users page.



# Checklist:
